src/main/java/org/cascadebot/cascadebot/music/KaraokeHandler.java

```
@@ -0,0 +1,115 @@
+                if (response.body() == null || response.code() != 200) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                String body = null;
+                try {
+                    body = response.body().string();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                if (body.equals("")) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                try {
+                    Document doc = finalBuilder.parse(new StringInputStream(body));
+                    NodeList texts = doc.getElementsByTagName("text");
+                    Captions captions = new Captions();
+                    for (int i = 0; i < texts.getLength(); i++) {
+                        NamedNodeMap attrs = texts.item(i).getAttributes();
+                        Node dur = attrs.getNamedItem("dur");
+                        Node start = attrs.getNamedItem("start");
+
+                        if (dur != null && start != null) {
+                            captions.addCaption(StringEscapeUtils.unescapeHtml(texts.item(i).getTextContent()), Double.parseDouble(start.getNodeValue()), Double.parseDouble(dur.getNodeValue()));
+                        }
+                    }
+
+                    new CaptionsTask(guildId, channel.getIdLong(), message.getIdLong(), captions).start(0, CAPTION_BUFFER_TIME * 1000);
+
+                } catch (IOException | SAXException | MDCException e) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    CascadeBot.LOGGER.error("Error in karaoke handler", e);
```

Same here as before, we shouldn't be swallowing the error since we shouldn't be getting these errors. Use the `sendException` method to report the error to the user so they can report it to us

Also why are you catching MDCException?

---

I'm pretty sure you told me to swallow the error in the first place, but alright.

Where's the sendException method?

---

MessagingTyped.replyException

---

MessagingTyped.replyException